Fix projection issue preventing selection of the field, when subquery has window function with another field (#12116)
crate/crate: docs/appendices/release-notes/unreleased.rst

.. stable branch. You can add a version label (`v/X.Y`) to the pull request for
.. an automated mergify backport.

> - Fixed an issue that caused a failure of the subquery with a windows function
>   over an unused field. For example,

Reviewer: ```suggestion
- Fixed an issue that caused a failure when a window function over a partition
   is not used in an upper query. For example:
```
